fastactivity.php

```diff
@@ -245,12 +245,15 @@ function fastactivity_civicrm_navigationMenu(&$menu) {
 function fastactivity_civicrm_links($op, $objectName, $objectId, &$links, &$mask, &$values) {
   $replace_search_links = (bool) CRM_Fastactivity_Settings::getValue('fastactivity_replace_search');
   if ($replace_search_links && $op == 'activity.selector.row') {
-    foreach ($links as &$link) {
-      if ($link['name'] == 'View') {
-        $link['url'] = 'civicrm/fastactivity/view';
-      }
-      elseif ($link['name'] == 'Edit') {
-        $link['url'] = 'civicrm/fastactivity/add';
+    // GP-31270 Wrong activity link (avoid replacement with fastactivity link when object leads to activity pages)
+    if ($objectName != 'Activity') {
```

Comment:
I don't think this makes sense - the goal is to not link to the fastactivity page if the original link doesn't link to the core activity page (but something like the contribution page instead). I don't think this can be determined by `$objectName` (as it'll always be `Activity`).

- Core behaviour for most activities: link to e.g. `/civicrm/contact/view/contribution...`
- Core behaviour for some activity types (like contribution): link to e.g. `/civicrm/contact/view/contribution...`

For the first scenario, we want to replace the link with the fastactivity link. For the second scenario, we don't want to do that (and keep the e.g. contribution view link instead).